src/pages/library/Showcase.js

```diff
@@ -16,104 +18,333 @@ import * as Style from 'src/libs/style'
 import * as Utils from 'src/libs/utils'
 
 
+// Description of the structure of the sidebar. Case is preserved when rendering but all matching is case-insensitive.
+const sidebarSections = [{
+  name: 'Getting Started',
+  labels: ['Workflow Tutorials', 'Notebook Tutorials', 'Data Tutorials', 'RStudio Tutorials', 'Galaxy Tutorials'],
+  keepCollapsed: true
+}, {
+  name: 'Analysis Tools',
+  labels: ['WDLs', 'Jupyter Notebooks', 'RStudio', 'Galaxy', 'Hail', 'Bioconductor', 'GATK', 'Cumulus', 'Spark']
+}, {
+  name: 'Experimental Strategy',
+  labels: ['GWAS', 'Exome Analysis', 'Whole Genome Analysis', 'Fusion Transcript Detection', 'RNA Analysis', 'Machine Learning',
+    'Variant Discovery', 'Epigenomics', 'DNA Methylation', 'Copy Number Variation', 'Structural Variation', 'Functional Annotation']
+}, {
+  name: 'Data Generation Technology',
+  labels: ['10x analysis', 'Bisulfate Sequencing']
+}, {
+  name: 'Scientific Domain',
+  labels: ['Cancer', 'Infectious Diseases', 'MPG', 'Single-cell', 'Immunology']
+}, {
+  name: 'Datasets',
+  labels: ['AnVIL', 'CMG', 'CCDG', 'TopMed', 'HCA', 'TARGET', 'ENCODE', 'BioData Catalyst', 'TCGA', '1000 Genomes', 'BRAIN Initiative',
+    'gnomAD', 'NCI', 'COVID-19']
+}, {
+  name: 'Utilities',
+  labels: ['Format Conversion', 'Developer Tools']
+}, {
+  name: 'Projects',
+  labels: ['HCA', 'AnVIL', 'BRAIN Initiative', 'BioData Catalyst', 'NCI']
+}]
+
+const uniqueSidebarTags = _.flow([
```

Comment:
Inline this?

Reply:
This is used by `groupByFeaturedTags` which operates on the current, possibly filtered, list of workspaces. I was worried that that computation was already a little heavier than I wanted. My intent here was to compute `uniqueSidebarTags` once ahead of time since that is only dependent on the static `sidebarSections`.

Do you still want me to inline it? Or is there a better way to express this? (I considered `_.once`, but that seemed unnecessary since there's no reason to defer the calculation.)

Reply:
I think there are a bunch of things that are being recomputed on every commit now, but I think a bunch of them may just need to be recomputed when the featured workspace list is loaded; I bet we could take care of that with some memoization...

Reply:
I wanted to do a better job with that, but unfortunately I just ran out of time.

I also couldn't make sense of the `StateHistory.update({ featuredWorkspaces })`, which looks like it wants to avoid repeated downloads, but there wasn't any condition guarding that fetch.

Anyway, I'm definitely open to adding some memoization. My first candidate is `groupByFeaturedTags`. Let me know if you have other suggestions.